cms/models/pagemodel.py

```diff
@@ -174,10 +175,15 @@ def get_absolute_url(self, language=None, fallback=True):
         if not language:
             language = get_language()
         with i18n.force_language(language):
-            if self.is_home:
-                return reverse('pages-root')
             path = self.get_path(language, fallback) or self.get_slug(language, fallback)
-            return reverse('pages-details-by-slug', kwargs={"slug": path})
+            if self.is_home:
+                relative_url = reverse('pages-root')
+            else:
+                relative_url = reverse('pages-details-by-slug', kwargs={"slug": path})
+            if self.site:
```

Comment:
This check can be optimized to reduce SQL queries when page's site is the same as configured in settings.

    if self.site_id and settings.SITE_ID != self.site_id and self.site.domain:
        return '//%s%s' % (self.site.domain, relative_url)

Actually I'm not sure if the check for site.domain is needed at all. `domain` field is not only required but also unique and if there is a `site_id` then there will be a `site.domain`.